packages/oas/medusa-oas-cli/src/command-oas.ts

```diff
@@ -0,0 +1,148 @@
+import * as path from "path"
+import { lstat, mkdir, writeFile } from "fs/promises"
+import swaggerInline from "swagger-inline"
+import OpenAPIParser from "@readme/openapi-parser"
+import { OpenAPIObject } from "openapi3-ts"
+import { Command, Option, OptionValues } from "commander"
+
+/**
+ * Constants
+ */
+// Medusa core package directory
+const medusaPackagePath = path.dirname(
+  require.resolve("@medusajs/medusa/package.json")
+)
+
+type ApiType = "store" | "admin"
+
+/**
+ * CLI Command declaration
+ */
+export const commandName = "oas"
+export const commandDescription =
+  "Compile full OAS from swagger-inline compliant JSDoc."
+
+export const commandOptions: Option[] = [
+  new Option("-t, --type <type>", "API type to compile []")
+    .choices(["all", "admin", "store"])
+    .default("all"),
+  new Option(
+    "-o, --out-dir <outDir>",
+    "Destination directory to output generated OAS files."
+  ).default(process.cwd()),
+  new Option("-D, --dry-run", "Do not output files."),
+  new Option(
+    "-p, --paths <paths...>",
+    "Additional paths to crawl for OAS JSDoc."
+  ),
+  new Option("-F, --force", "Ignore OAS validation and output OAS files."),
+]
+
+export function getCommand() {
+  const command = new Command(commandName)
+  command.description(commandDescription)
+  for (const opt of commandOptions) {
+    command.addOption(opt)
+  }
+  command.action(async (options) => await execute(options))
+  return command
+}
+
+/**
+ * Main
+ */
+export async function execute(cliParams: OptionValues) {
+  /**
+   * Process CLI options
+   */
+  const dryRun = !!cliParams.dryRun
+  const force = !!cliParams.force
+
+  const apiTypesToExport =
+    cliParams.type === "all" ? ["store", "admin"] : [cliParams.type]
+
+  const outDir = path.resolve(cliParams.outDir)
+  if (!(await lstat(outDir)).isDirectory()) {
+    throw new Error(`--outDir must be a directory - ${outDir}`)
+  }
+
+  const additionalPaths = (cliParams.paths ?? []).map((additionalPath) =>
+    path.resolve(additionalPath)
+  )
+  for (const additionalPath of additionalPaths) {
+    if (!(await lstat(additionalPath)).isDirectory()) {
```

Comment:
suggestion: Same as the previous one

Reply:
Slight spin on the suggestion. I created a helper method instead to abstract the complexity of `lstat`. See 9bc39d1b9af81567f9f92b4db72f83633ceb2e70